python/cudf/cudf/pylibcudf_tests/test_string_find.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import numpy as np
+import pyarrow as pa
+import pytest
+from utils import assert_column_eq
+
+import cudf._lib.pylibcudf as plc
+from cudf._lib.scalar import DeviceScalar
+
+
+@pytest.fixture()
+def plc_col():
+    data = pa.array(
+        ["AbC", "de", "FGHI", "j", "kLm", "nOPq", None, "RsT", None, "uVw"]
+    )
+    return plc.interop.from_arrow(data)
+
+
+@pytest.fixture
+def find_target_column():
+    return plc.interop.from_arrow(
+        pa.array(["A", "d", "F", "j", "k", "n", None, "R", None, "u"])
+    )
+
+
+@pytest.fixture
+def contains_target_column():
+    return plc.interop.from_arrow(
+        pa.array(["a", "d", "F", "j", "m", "q", None, "R", None, "w"])
+    )
+
+
+@pytest.fixture
+def starts_with_target_column():
+    return plc.interop.from_arrow(
+        pa.array(["A", "d", "F", "j", "k", "n", None, "R", None, "u"])
+    )
+
+
+@pytest.fixture
+def ends_with_target_column():
+    return plc.interop.from_arrow(
+        pa.array(["C", "e", "I", "j", "m", "q", None, "T", None, "w"])
+    )
+
+
+@pytest.mark.parametrize("target", ["a", ""])
+def test_find(plc_col, target):
+    got = plc.strings.find.find(
+        plc_col, DeviceScalar(target, dtype=np.dtype("object")).c_value, 0, -1
+    )
+
+    expected = pa.array(
+        [
+            elem.find(target) if elem is not None else None
+            for elem in plc.interop.to_arrow(plc_col).to_pylist()
```

Comment:
If you continued defining the `strings_col` fixture that you originally had and made the `plc_col` fixture depend on that fixture, you could have all of these tests take both of them as arguments and not need extra `to_arrow` calls. Does that make sense? Basically like this:
```
@fixture
def strings_col(...):
    ...

@fixture
def plc_col(strings_col):
    return plc.interop.from_arrow(strings_col)

def test_*(plc_col, strings_col):
```